File: dataModule/src/main/java/com/mobnetic/coinguardian/model/market/Vaultoro.java
package com.mobnetic.coinguardian.model.market;

import java.util.HashMap;
import java.util.LinkedHashMap;

import com.mobnetic.coinguardian.model.CheckerInfo;
import com.mobnetic.coinguardian.model.Market;
import com.mobnetic.coinguardian.model.Ticker;
import com.mobnetic.coinguardian.model.currency.Currency;
import com.mobnetic.coinguardian.model.currency.VirtualCurrency;

public class Vaultoro extends Market {

	private final static String NAME = "Vaultoro";
	private final static String TTS_NAME = NAME;
	private final static String URL = "https://api.vaultoro.com/latest/";
	private final static HashMap<String, CharSequence[]> CURRENCY_PAIRS = new LinkedHashMap<String, CharSequence[]>();
	static {
		CURRENCY_PAIRS.put(Currency.GOLD, new String[]{
				VirtualCurrency.BTC
			});
	}
	
	public Vaultoro() {
		super(NAME, TTS_NAME, CURRENCY_PAIRS);
	}
	
	@Override
	public String getUrl(int requestId, CheckerInfo checkerInfo) {
		return URL;
	}
	
	@Override
	protected void parseTicker(int requestId, String responseString, Ticker ticker, CheckerInfo checkerInfo) throws Exception {
		ticker.last = Double.parseDouble(responseString); 
	}
}

File: dataModule/src/main/java/com/aneonex/bitcoinchecker/datamodule/model/market/Vaultoro.kt
package com.aneonex.bitcoinchecker.datamodule.model.market

import com.aneonex.bitcoinchecker.datamodule.exceptions.MarketParseException
import com.aneonex.bitcoinchecker.datamodule.model.CheckerInfo
import com.aneonex.bitcoinchecker.datamodule.model.Market
import com.aneonex.bitcoinchecker.datamodule.model.Ticker
import com.aneonex.bitcoinchecker.datamodule.model.currency.Currency
import com.aneonex.bitcoinchecker.datamodule.model.currency.CurrencyPairsMap
import com.aneonex.bitcoinchecker.datamodule.model.currency.VirtualCurrency
import org.json.JSONObject
import java.util.concurrent.TimeUnit

class Vaultoro : Market(NAME, TTS_NAME, CURRENCY_PAIRS) {
    companion object {
        private const val NAME = "Vaultoro"
        private const val TTS_NAME = NAME
        private const val URL = "https://api.vaultoro.com/v1/public/ohlc?pair=%1\$s&since=%2\$s"
        private val CURRENCY_PAIRS: CurrencyPairsMap = CurrencyPairsMap()

        init {
            CURRENCY_PAIRS[Currency.GOLD] = arrayOf(
                    VirtualCurrency.BTC,
                    VirtualCurrency.DASH
            )
        }
    }

    override fun getUrl(requestId: Int, checkerInfo: CheckerInfo): String {
        var pairId = checkerInfo.currencyPairId
        if(pairId == null){
            pairId = "${checkerInfo.currencyBase}${checkerInfo.currencyCounter}"
        }

        val sinceLastWeekTime = (System.currentTimeMillis() - TimeUnit.DAYS.toMillis(7)) / 1000
        return String.format(URL, pairId, sinceLastWeekTime)
    }

    @Throws(Exception::class)
    override fun parseTickerFromJsonObject(requestId: Int, jsonObject: JSONObject, ticker: Ticker, checkerInfo: CheckerInfo) {
        val dataJson = jsonObject.getJSONArray("data")
        if(dataJson.length() == 0) throw MarketParseException("No data")

        val tickerJson = dataJson.getJSONObject(0)
        ticker.last = tickerJson.getDouble("close")
        ticker.high = tickerJson.getDouble("high")
        ticker.low = tickerJson.getDouble("low")
        ticker.vol = tickerJson.getDouble("volume")
        ticker.timestamp = tickerJson.getLong("createdAt") * 1000
    }
}